subsys/net/ip/net_mgmt.c

```diff
@@ -56,7 +56,7 @@ static inline void mgmt_push_event(u32_t mgmt_event, struct net_if *iface,
 	ARG_UNUSED(length);
 #endif /* CONFIG_NET_MGMT_EVENT_INFO */
 
-	k_sem_take(&net_mgmt_lock, K_FOREVER);
+	k_sem_take(&net_mgmt_lock, k_is_in_isr() ? K_NO_WAIT : K_FOREVER);
```

Comment:
@tbursztyka, What it k_sem_take() fails with K_NO_WAIT?

Reply:
can you comment on the original pr #8112 instead?

Reply:
Done: https://github.com/zephyrproject-rtos/zephyr/pull/8112/files#r192724235